refinery/ui/source/js/dashboard/controllers/analysis-deletion.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Created by scott on 9/28/16.
+ */
+
+'use strict';
+
+function AnalysisDeleteCtrl (
+  $log,
+  $uibModalInstance,
+  _,
+  deletionService,
+  config,
+  analysis,
+  analyses,
+  analysesReloadService,
+  isOwner,
+  dashboardDataSetsReloadService
+) {
+  this.$log = $log;
+  this._ = _;
+  this.config = config;
+  this.analysis = analysis;
+  this.analyses = analyses;
+  this.$uibModalInstance = $uibModalInstance;
+  this.deletionService = deletionService;
+  this.analysesReloadService = analysesReloadService;
+  this.isOwner = isOwner;
+  this.dashboardDataSetsReloadService = dashboardDataSetsReloadService;
+}
+
+/**
+ * Cancel object deletion.
+ * @type  {function}
+ */
+AnalysisDeleteCtrl.prototype.cancel = function () {
+  this.deletionMessage = null;
+  this.$uibModalInstance.dismiss('cancel');
+};
+
+/**
+ * Delete an Analysis
+ * @type   {function}
+ */
+AnalysisDeleteCtrl.prototype.delete = function () {
+  var that = this;
+  that.isDeleting = true;
+  that.deletionMessage = null;
+  that.deleteSuccessful = false;
+
+  this
+    .deletionService
+    .delete({
+      model: this.config.model,
+      uuid: this.config.uuid
+    })
+    .$promise
+    .then(function (response) {
+      that.deletionMessage = response.data;
+      that.isDeleting = false;
+      if (response.status === 200) {
```

Comment:
Conditional not needed. The then successful call back will catch 200s otherwise the error callback will be called.